packages/core/src/core/sunrise-sunset-calculator.ts

```diff
@@ -0,0 +1,250 @@
+  };
+
+  /**
+   * Calculate sunrise and sunset times for a given date
+   * @param date - The date to calculate times for
+   * @param calendar - The calendar definition
+   * @returns Object with sunrise and sunset as hour decimal values (e.g., 6.5 = 6:30)
+   */
+  static calculate(
+    date: CalendarDate,
+    calendar: SeasonsStarsCalendar
+  ): { sunrise: number; sunset: number } {
+    if (!calendar.seasons || calendar.seasons.length === 0) {
+      return this.getDefaultTimes(calendar);
+    }
+
+    // Find current season and next season
+    const currentSeasonIndex = this.findSeasonIndex(date, calendar);
+    if (currentSeasonIndex === -1) {
+      return this.getDefaultTimes(calendar);
+    }
+
+    const currentSeason = calendar.seasons[currentSeasonIndex];
+    const nextSeasonIndex = (currentSeasonIndex + 1) % calendar.seasons.length;
+    const nextSeason = calendar.seasons[nextSeasonIndex];
+
+    // Get sunrise/sunset for current and next season
+    const currentTimes = this.getSeasonTimes(currentSeason, calendar);
+    const nextTimes = this.getSeasonTimes(nextSeason, calendar);
+
+    // Calculate progress through the season (0 = first day, 1 = last day)
+    const progress = this.calculateSeasonProgress(date, currentSeason, nextSeason, calendar);
+
+    // Interpolate between current and next season
+    return {
+      sunrise: this.interpolate(currentTimes.sunrise, nextTimes.sunrise, progress),
+      sunset: this.interpolate(currentTimes.sunset, nextTimes.sunset, progress),
+    };
+  }
+
+  /**
+   * Find the index of the season that contains the given date
+   */
+  private static findSeasonIndex(date: CalendarDate, calendar: SeasonsStarsCalendar): number {
+    for (let i = 0; i < calendar.seasons.length; i++) {
+      if (this.isDateInSeason(date, calendar.seasons[i], calendar)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Check if a date falls within a season
+   */
+  private static isDateInSeason(
+    date: CalendarDate,
+    season: CalendarSeason,
+    calendar: SeasonsStarsCalendar
+  ): boolean {
+    const month = date.month;
+    const day = date.day;
+    const startMonth = season.startMonth;
+    const endMonth = season.endMonth ?? season.startMonth;
+    const startDay = season.startDay ?? 1;
+    const endDay = season.endDay ?? (calendar.months?.[endMonth - 1]?.days ?? 31);
+
+    // Year-crossing season (e.g., Winter: Dec-Feb)
+    if (startMonth > endMonth) {
+      return (
+        month > startMonth ||
+        month < endMonth ||
+        (month === startMonth && day >= startDay) ||
+        (month === endMonth && day <= endDay)
+      );
+    }
+
+    // Regular season
+    if (month < startMonth || month > endMonth) {
+      return false;
+    }
+    if (month === startMonth && day < startDay) {
+      return false;
+    }
+    if (month === endMonth && day > endDay) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
I think this is implemented already in a utility module of some sort. Look for a pre-existing version of this function and use that instead.